specs/lightclient/validator.md

```diff
@@ -0,0 +1,395 @@
+
+```python
+def construct_sync_committee_contribution(state: BeaconState, sync_committee_signatures: Set[SyncCommitteeSignature]) -> SyncCommitteeContribution:
+    contribution = SyncCommitteeContribution()
+    contribution.slot = state.slot
+
+    sync_committee = state.current_sync_committee
+    epoch = compute_epoch_at_slot(state.slot)
+    domain = get_domain(state, DOMAIN_SYNC_COMMITTEE, epoch)
+    signing_root = compute_signing_root(
+      get_block_root_at_slot(state, state.slot), 
+      domain
+    )
+
+    signatures = []
+    for sync_committee_signature in sync_committee_signatures:
+        index = sync_committee_signature.index
+        signature = sync_committee_signature.signature
+        contribution.aggregate_bits[index] = True
+        signatures.append(signature)
+
+        # is sync committee signature valid?
+        assert sync_committee_signature.slot == state.slot
+        assert bls.Verify(
+          sync_committee.pubkeys[index], 
+          signing_root,
+          signature
+        )
+
+    contribution.signature = bls.Aggregate(signatures)
+    return contribution
+```
+
+##### Broadcast sync committee contribution
+
+If the validator is selected to aggregate (`is_sync_committee_aggregator`), then they broadcast their best aggregate as a `SignedContributionAndProof` to the global aggregate channel (`sync_committee_contribution_and_proof` topic) two-thirds of the way through the `slot`-that is, `SECONDS_PER_SLOT * 2 / 3` seconds after the start of `slot`.
+
+Selection proofs are provided in `ContributionAndProof` to prove to the gossip channel that the validator has been selected as an aggregator.
+
+`ContributionAndProof` messages are signed by the aggregator and broadcast inside of `SignedContributionAndProof` objects to prevent a class of DoS attacks and message forgeries.
+
+First, `contribution_and_proof = get_contribution_and_proof(state, validator_index, contribution, privkey)` is constructed.
+
+```python
+def get_contribution_and_proof(state: BeaconState,
+                               aggregator_index: ValidatorIndex,
+                               contribution: SyncCommitteeContribution,
+                               privkey: int) -> ContributionAndProof:
+    selection_proof = get_sync_committee_slot_signature(
+      state, 
+      contribution.slot,
+      privkey
+    )
+    return ContributionAndProof(
+        aggregator_index=aggregator_index,
+        contribution=contribution,
+        selection_proof=selection_proof,
+    )
+```
+
+Then `signed_contribution_and_proof = SignedContributionAndProof(message=contribution_and_proof, signature=signature)` is constructed and broadast. Where `signature` is obtained from:
+
+```python
+def get_contribution_and_proof_signature(
+  state: BeaconState,
```

Comment:
spacing. prefer 4-spaces and prefer trailing off the function name like above (`get_contribution_and_proof`)